integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/SqlDWDatabricksVisitor.java

```diff
@@ -0,0 +1,114 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.api.DatasetFactory;
+import io.openlineage.spark.api.OpenLineageContext;
+import io.openlineage.spark.api.QueryPlanVisitor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.commons.lang3.reflect.MethodUtils;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.sources.BaseRelation;
+
+/**
+ * {@link LogicalPlan} visitor that matches SqlDWRelation that comes from an Azure Databricks
+ * environment. This function extracts a {@link OpenLineage.Dataset} from the SQL DW/ Synapse table
+ * referenced by the relation. The convention used for a namespace is a URI of <code>
+ * sqlserver://&lt;server&gt;.&lt;.datasetId&gt;.&lt;tableName&gt;</code> . The name for Sql Dw
+ * tables may be table name (e.g. "exampleInputA") or a multi-part name (e.g.
+ * "[dbo].[exampleInputA]"). If the data source is a query (e.g. <code>
+ * ((select \"id\" FROM dbo.exampleInputA WHERE postalCode != '55555') q)</code>) then the name will
+ * be COMPLEX.
+ */
+public class SqlDWDatabricksVisitor<D extends OpenLineage.Dataset>
+    extends QueryPlanVisitor<LogicalPlan, D> {
+  private final DatasetFactory<D> factory;
+  private static final Pattern dbJdbcPattern = Pattern.compile("database=([^;]*);?");
+  private static final Pattern serverJdbcPattern = Pattern.compile("jdbc:([^;]*);?");
+  private static final String DATABRICKS_CLASS_NAME = "com.databricks.spark.sqldw.SqlDWRelation";
+
+  public SqlDWDatabricksVisitor(OpenLineageContext context, DatasetFactory<D> factory) {
+    super(context);
+    this.factory = factory;
+  }
+
+  public static boolean hasSqlDWDatabricksClasses() {
+    try {
+      SqlDWDatabricksVisitor.class.getClassLoader().loadClass(DATABRICKS_CLASS_NAME);
+      return true;
+    } catch (Exception e) {
+      // swallow- we don't care
+    }
+    return false;
+  }
+
+  protected boolean isSqlDwRelationClass(LogicalPlan plan) {
+    return plan instanceof LogicalRelation
+        && ((LogicalRelation) plan).relation().getClass().getName().equals(DATABRICKS_CLASS_NAME);
+  }
+
+  @Override
+  public boolean isDefinedAt(LogicalPlan plan) {
+    return isSqlDwRelationClass(plan);
+  }
+
+  private String getName(BaseRelation relation) {
+    String tableName;
+    try {
+      tableName = (String) FieldUtils.readField(relation, "tableNameOrSubquery", true);
+    } catch (IllegalAccessException e) {
+      throw new RuntimeException("Unable to discover SQLDW tableNameOrSubquery property");
```

Comment:
I would not throw an exception here. Exceptions can cause the overall lineage event construction to fail (there should be try/catches, but...), but really only this component is missing some details. I'd log a warning and return an empty list from the `apply` method if we can't get the information we need.

Reply:
Got it!  I had seen a few other visitors actually throw an exception but I totally agree with and understand this intention. Thank you for explaining.